### ISSUE TITLE
IPF Support

### ISSUE THREAD

**Issue by Enverex** [libretro/hatari#4]
I've installed the IPF libraries/headers where they state they should be installed but Hatari still claims it was built without IPF support. I'm not seeing a configure switch for this so I can only assume it should be picked up automatically.

Does/should IPF support work on the LibRetro version?

**Comment by r-type** [comment]
>Does/should IPF support work on the LibRetro version?
in short , for now  with current makefile, no.

If you want IPF support , you have to tweak makefile.libretro to enable it.
basically to set HAVE_CAPSIMAGE to 1and deal with the IPF include and lib dir.

I have tested with the 4.2 user distribution for linux found here http://www.softpres.org/download
i tweak the makefile like here http://pastebin.com/MSs8k4LV
and build with 
```
make -f Makefile.libretro "IPFSUPPORT=1" "IPFDIR=../x86_64-linux-gnu-capsimage"
```
then IPF load fine  .

**Comment by Enverex** [comment]
Does it compile fine (albeit without IPF support) if you tell it to compile with IPF support but are missing the libraries? The reason I ask is this should really be on by default (but leave it up to the user to install those libraries themselves).

The IPF disk images are now the defacto for Atari ST it seems, so not including support to actually read them seems like a rather massive oversight.

**Comment by r-type** [comment]
I only post you a quick way to handle it for yourself if you need IPF support. 
Now if someone want to handle IPF support in a proper way ...

> The IPF disk images are now the defacto for Atari ST it seems, so not including support to actually read them seems like a rather massive oversight.

This port is outdated  and most user have msa / st / or stx image (like in TOSEC )

**Comment by Enverex** [comment]
TOSEC has always been a bit of a mess (no-one needs 20 copies of games, especially broken or bad copies). The cleanest versions of Atari ST games available these days are from No-Intro and they only release IPF files from what I can tell, hence the question regarding support.

**Comment by r-type** [comment]
Notes that i m not against the IPF , as it  appearing to me to be the best choice from preservation point of view .

I only said that most user have msa or st img (as they are easier to find ) , so they don't probably don't mind about IPF.

Also notes that personally i like more the intro than the games themselves  , maybe i'm a bit notsalgic
re-viewing all the atari crew ...

BTW when i ll have some free time , i plan to redone a port more up-to-date .

**Comment by Enverex** [comment]
No worries, I was just concerned as I'd seem more IPFs than ST/STX files around lately so it seemed like something that was needed going forward.

I agree on the cracktro front, but I take a nicer collection over a TOSEC mess any day :P

**Comment by joolswills** [comment]
Thanks. Would be useful to have the basic IPF support changed to Makefile.libretro committed to the repo.

**Comment by M1kerochip** [comment]
Is this ever going to be fixed?

**Comment by ToniBC** [comment]
I guess when someone gets the core, check it and update, it will surely be put on the support. This, like some others, remained stagnant in the initial ports and little was reviewed, I suppose due to lack of interest in these systems.

**Comment by mistepien** [comment]
IMHO the best option is to clone this repository here https://github.com/simonowen/capsimage
and compile all together. Capsimage5.1 is compatible with gcc9 and the issue would be fully resolved.

**Comment by r-type** [comment]
IMHO , the best option is to rebase/replace core with last hatari git version before change/add anythings here.

BTW, the ipf support in hatari-libretro work from years,it is just that the commit has not been push here!

**Comment by mspykerez** [comment]
> IMHO , the best option is to rebase/replace core with last hatari git version before change/add anythings here.
> 
> BTW, the ipf support in hatari-libretro work from years,it is just that the commit has not been push here!

+1

**Comment by pkos** [comment]
If you are looking to compile your own Hatari (Atari - ST) core with support for IPF files, I've got the code:
https://github.com/pkos/hatari

Unfortunately, RetroArch cannot distribute the Hatari core with IPF support due to GLP license.

**Comment by TiBeN** [comment]
Any news on this ? 

A PR exists but have not been merged:
<https://github.com/libretro/hatari/pull/55>

**Comment by WizzardSK** [comment]
I think we can close this.

**Comment by Shoegzer** [comment]
I don't think this should be closed. How does the core build the `libcapsimage.so` library ~and then build the core with IPF support in linux~?

There is a way to build the library from @r-type's repository [here](https://github.com/r-type/hataritmp/tree/master/libretro/capsimg_linux), but that's not part of the official libretro repo and has to be done manually.

Update: Actually, the core DOES build with IPF support in linux if the library is detected and works fine, so nevermind that part. If building the library itself is out of scope for this PR, then disregard, but it would be nice to put capsimg in externals and build from that if it isn't detected.

**Comment by cscd98** [comment]
@Shoegzer The new core can be built with the following options which builds capsimage for you if not installed:

```
-DENABLE_LIBRETRO=ON -DENABLE_HATARI=OFF -DENABLE_TOOLS=OFF -DENABLE_STATIC_ZLIB=ON -DENABLE_STATIC_CAPSIMAGE=ON
```

**Comment by Shoegzer** [comment]
Ah, thanks. I assumed it would have been in `CMakeLists.txt` so I didn't look anywhere else, but it was actually [here](https://github.com/libretro/hatari/blob/main/.gitlab-ci-libretro.yml).

BTW, puae is the only other core that uses capsimage iirc. Since that core doesn't use CMake there are no CMake flags, but I think it builds and uses capsimage by default anyway?

**Comment by cscd98** [comment]
It was added into here: https://github.com/libretro/hatari/blob/main/cmake/BuildDependencies.cmake to make it cleaner.

capsimg doesn't build for all platforms, that code needs some fixing up.

**Comment by Shoegzer** [comment]
> It was added into here: https://github.com/libretro/hatari/blob/main/cmake/BuildDependencies.cmake to make it cleaner.

Ah, got it.  Yeah that's better.

> capsimg doesn't build for all platforms, that code needs some fixing up.

It does look somewhat inconsistent.  I just built the puae core for linux-x64 and it seemed to build, but I can't be sure it works without an IPF image handy to test.  For now I'll assume it does.
